cpp/src/arrow/CMakeLists.txt

```diff
@@ -402,6 +402,7 @@ list(APPEND
      compute/registry.cc
      compute/kernels/codegen_internal.cc
      compute/kernels/row_encoder.cc
+     compute/kernels/ree_util_internal.cc
```

Comment:
Does this need to be included in the default build or should it be conditionally compiled when `ARROW_COMPUTE=ON` alongside `vector_run_end_encode.cc` in [this block](https://github.com/apache/arrow/blob/3a3d1ccde4023cb769eb8a2c10531ee6dce666dc/cpp/src/arrow/CMakeLists.txt#L428)? If you expect the utils will be used outside of the REE-kernels then it's probably fine where it is.

Reply:
`RunEndEncode` is extremely useful when writing unit tests, so I would consider this "core" and `RunEndDecode` will probably play a central role in `compute` as something we can reach out to whenever a function can't handle REE directly.

Reply:
Alright, sounds good. Just wanted to make sure that it was intentional.